### DISCUSSION TITLE
[Bug]: Declaration of Pimcore\Bundle\ApplicationLoggerBundle\ApplicationLogger::emergency(Stringable|string $message, array $context = []): void must be compatible with Psr\Log\LoggerInterface::emergency($message, array $context = [])

### DISCUSSION THREAD

**Discussion by bitbirddev** [pimcore/pimcore#16254] [Q&A / Support]
### Pimcore version

v11.1.0

### Steps to reproduce

run `composer dump-env`

since v11.1.0 the error `Fatal error: Declaration of Pimcore\Bundle\ApplicationLoggerBundle\ApplicationLogger::emergency(Stringable|string $message, array $context = []): void must be compatible with Psr\Log\LoggerInterface::emergency($message, array $context = []) in ./vendor/pimcore/pimcore/bundles/ApplicationLoggerBundle/src/ApplicationLogger.php on line 224` gets thrown. its still working with Pimcore Version v11.0.12

### Actual Behavior

Error gets thrown 

`Fatal error: Declaration of Pimcore\Bundle\ApplicationLoggerBundle\ApplicationLogger::emergency(Stringable|string $message, array $context = []): void must be compatible with Psr\Log\LoggerInterface::emergency($message, array $context = []) in ./vendor/pimcore/pimcore/bundles/ApplicationLoggerBundle/src/ApplicationLogger.php on line 224`

### Expected Behavior

.env gets compiled to .env.local.php

**Comment by jdreesen** [comment]
This is strange, because it should only happen if you have `psr/log` `v1` installed (which should not be possible, since `monolog/monolog` in `^3.2.0` requires at least `v2`).

Which version of `psr/log` do you have installed?
please run: `composer show psr/log`
and if it says `v1`, run: `composer why psr/log`

**Comment by bitbirddev** [comment]
its very strange indeed

- psr/log is 3.0.0
- also composer global show psr/log is 3.0.0
- i do not have installed the psr pecl extension
- composer version is 2.6.5

i have no idea to be honest. it happens locally on my dev machine and also while deploying to prod...

**Comment by dvesh3** [comment]
Hi @bitbirddev psr/log v3 should have the [type hints](https://github.com/php-fig/log/blob/3.0.0/src/LoggerInterface.php#L30) defined. Please check your vendor files and try clearing cache if the type hints are there.

**Comment by Fahl-Design** [comment]
FYI @dvesh3 @bitbirddev it not a pimcore problem. In my case it's caused a composer plugin "phpro/grumphp" try to disable/remove all "allow-plugins" from compsoer.json and just run `composer` without any params to see if it works.
to verify is caused by a plugin, run `composer --no-plugins`

**Comment by bitbirddev** [comment]
@Fahl-Design composer runs without any problems, its just the "composer dump-env prod" command that gives me this error. this happens on my local machine, on github actions with shivammathur/setup-php and also on my ubuntu production server.

**Comment by Fahl-Design** [comment]
@bitbirddev in this case try to add `--no-plugins` to it and give it a try or maybe check the error stack trace with `-vvv` first.

**Comment by langert** [comment]
I got the same problem and could solve it with calling bin/console dotenv:dump without the error coming up.

**Comment by MarcoLeko** [comment]
We are encountering the same error now when migrating from Pim 10 to v11. And yeah the strange thing is the LoggerInterface is indeed type hinted and correctly implemented :/

**Reply by bitbirddev** [reply]
i've never found a solution for that. please let me know if you find one :)
